src/ctia/entity/judgement/schemas.clj

```diff
@@ -37,42 +38,34 @@
 (s/defschema PartialStoredJudgement
   (csu/optional-keys-schema StoredJudgement))
 
-(s/defn realize-judgement :- StoredJudgement
-  ([new-judgement :- NewJudgement
-    id :- s/Str
-    tempids :- (s/maybe TempIDs)
-    owner :- s/Str
-    groups :- [s/Str]]
+(def judgement-default-realize
+  (default-realize-fn "judgement" NewJudgement StoredJudgement))
+
+(s/defn realize-judgement :- (with-error StoredJudgement)
+  ([new-judgement id tempids owner groups]
    (realize-judgement new-judgement id tempids owner groups nil))
   ([new-judgement :- NewJudgement
     id :- s/Str
     tempids :- (s/maybe TempIDs)
     owner :- s/Str
     groups :- [s/Str]
-    prev-object :- (s/maybe StoredJudgement)]
-   (let [now (time/now)
-         disposition (try
-                       (determine-disposition-id new-judgement)
-                       (catch clojure.lang.ExceptionInfo _
-                         (throw
-                          (http-response/bad-request!
-                           {:error "Mismatching :dispostion and dispositon_name for judgement"
-                            :judgement new-judgement}))))
-         disposition_name (get disposition-map disposition)]
-     (merge new-judgement
-            {:id id
-             :type "judgement"
-             :disposition disposition
-             :disposition_name disposition_name
-             :owner (or (:owner prev-object) owner)
-             :groups (or (:groups prev-object) groups)
-             :created (or (:created prev-object) now)
-             :tlp (:tlp new-judgement
-                        (:tlp prev-object (properties-default-tlp)))
-             :schema_version schema-version}
-            (make-valid-time (:valid_time prev-object)
-                             (:valid_time new-judgement)
-                             now)))))
+    prev-judgement :- (s/maybe StoredJudgement)]
+   (let [{:keys [error] :as disposition}
+         (try
+           (determine-disposition-id new-judgement)
+           (catch clojure.lang.ExceptionInfo e
```

Comment:
Wouldn't it be safe to check the `ex-data` of the `ExceptionInfo` here to be certain this is a Mismatching disposition problem at the source of the exception?

Reply:
Yes it would be better but it was not in the scope of this PR